x-pack/solutions/search/plugins/search_homepage/public/components/getting_started_redirect_gate.tsx

```diff
@@ -16,13 +16,36 @@ interface Props {
 }
 
 export const GettingStartedRedirectGate = ({ coreStart, children }: Props) => {
-  const isGettingStartedEnabled = useSearchGettingStartedFeatureFlag();
+  const isFeatureFlagEnabled = useSearchGettingStartedFeatureFlag();
+  const [userRoles, setUserRoles] = useState<string[]>([]);
+  const [hasCheckedRole, setHasCheckedRole] = useState(false);
+  const hasRedirected = useRef(false);
+
+  useEffect(() => {
+    // Get user role
+    coreStart.userProfile.getCurrent().then((userProfile) => {
+      const roles = userProfile?.user.roles || [];
+      setUserRoles([...roles]); // Spread to convert readonly array to mutable
```

Comment:
Q: does it not work otherwise 🤔

I assume we set the array to the readonly one, and if corestart changes, we try to re assign another user roles array here.